### DISCUSSION TITLE
Help migrating from 0.1.x to 0.2.x?

### DISCUSSION THREAD

**Discussion by alanbchristie** [Textualize/textual#1005] [Q&A] [answered]
OK - so it's not at version `1.x` yet so I guess that means everything can break at any time. That's SerVer for you!

Anyway, I have working console app based on `0.1.18` but it's broken under `0.2.1`. The first problem is...

```
TypeError: App.run() got an unexpected keyword argument 'title'
```

Without asking for specific help is there a _migration guide_? Something for those that have a working app in 0.1.18, not interested in CSS, but need the app working under 0.2.

Or is 0.2 just  a couple re-write and anything under 0.1 just won't work?

**Comment by davep** [comment, marked as answer]
There isn't a migration guide, at least not at the moment. In the case of the particular error you have there: the `title` keyword argument was removed in favour of a `TITLE` class property. See [here for example](https://github.com/Textualize/textual/blob/f3105d149c0f33991d9916a76eff9d7436d713f4/examples/five_by_five.py#L320).

**Comment by alanbchristie** [comment]
Thanks - I guess other errors I encounter really indicate I either need to embrace the new world and re-write the code (no views/docks) and instead read the example and reconstruct the code using 0.2's  style.

i.e. you can't expect to use a `0.1` app with the `0.2` codebase.

**Reply by davep** [reply]
That may need @willmcgugan or @darrenburns to answer with confidence -- I've only come to Textual during the "CSS branch era", as it were, so couldn't say with confidence. But do feel free to ask away on any issues you run in to; happy to help get things going.

**Comment by alanbchristie** [comment]
Having spent a short time on switching to 0.2 I think I'll revert my changes and return to 0.1 (views, docks, widgets). Nothing against what I'm sure is good stuff - but I'll have to come back to this when I have more time.

Thnaks.
